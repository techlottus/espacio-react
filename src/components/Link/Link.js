import React, { useEffect } from 'react'


export const Link = ({data, onClick})  => {
    const linkRef = React.createRef();

    useEffect(() => {
        linkRef.current.data = {
            text: data.text || '',
            size: data.size || '',
            isBold: data.isBold || false,
            disabled: data.disabled || false,
            id: data.id || '',
            icon: data.icon || '',

        }
    },[data]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        linkRef.current.addEventListener('onClick', onClick);
    },[]) // eslint-disable-line react-hooks/exhaustive-deps

    return(
        <>
        <lottus-link ref={linkRef}></lottus-link>
        </>
    )
}