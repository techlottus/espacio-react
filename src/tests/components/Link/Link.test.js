import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Link } from "../../../components/Link/Link";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState;
let data = {
  text: "",
  size: "",
  isBold: false,
  disabled: false,
  id: "",
  icon: "",
};
let onClick = jest.fn();

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Link data={data} onClick={onClick} />
    </Provider>
  );
};

describe("Test <Link/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
