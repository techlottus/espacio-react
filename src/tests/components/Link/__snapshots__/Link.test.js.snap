// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Link/> should init component 1`] = `<lottus-link />`;
